src/Microsoft.AspNetCore.ResponseCaching/Internal/MemoryCachedResponse.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.ResponseCaching.Internal
+{
+    internal class MemoryCachedResponse
+    {
+        public DateTimeOffset Created { get; set; }
+
+        public int StatusCode { get; set; }
+
+        public IHeaderDictionary Headers { get; set; } = new HeaderDictionary();
+
+        public List<byte[]> Shards { get; set; }
```

Comment:
Will need to come up with a consistent terminology since the naming is exposed via options. I might leave the pooling optimization to another PR. This one is already getting laggy on github.